src/Core/Util/File/FileSizeConverter.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Util\File;
+
+/**
+ * Converts int value to formatted file size value
+ */
+class FileSizeConverter
+{
+    /**
+     * @param int $bytes
+     * @param int $precision
+     *
+     * @return string
+     */
+    public function convert(int $bytes, int $precision = 2): string
+    {
+        $units = ['B', 'kB', 'MB', 'GB', 'TB'];
+        $bytes = max($bytes, 0);
+        $pow = floor(($bytes ? log($bytes) : 0) / log(1024));
+        $pow = min($pow, count($units) - 1);
+
+        if ($bytes >= 1024) {
+            $bytes /= pow(1024, $pow);
+            $bytes = number_format(round($bytes, $precision), 2, '.', '');
```

Comment:
The number should be formatting depending on cldr implementation? ping @matks

Reply:
This is a FileSizeConverter 🤔 do you think file size formats differ from one country to another ?

Reply:
The file format no, but the number format yes. It's the same for liter, meters, kilograms, etc. :) Maybe we should take in consideration this for later. Let's see with @eternoendless